makefiles/dependency_resolution.inc.mk

```
@@ -50,4 +50,10 @@ else
     $(shell $(COLOR_ECHO) "$(COLOR_RED)Deprecated modules are in use:$(COLOR_RESET)"\
                           "$(DEPRECATED_MODULES_USED)" 1>&2)
   endif
+
+  # Warn about telnet
+  ifneq (,$(filter auto_init_telnet,$(USEMODULE)))
+    $(shell $(COLOR_ECHO) "$(COLOR_YELLOW)Telnet will be started automatically,"\
+                          "don't run this on public networks!$(COLOR_RESET)" 1>&2)
+  endif
```

Maybe we can have this a bit more face jumping using red color. And maybe also ask the user to set `I_UNDERSTAND_THAT_TELNET_IS_INSECURE` to `1`, so that people using CIs and not looking at logs won't do stupid things.

```suggestion
  ifneq (,$(filter auto_init_telnet,$(USEMODULE)))
    ifneq (1,$(I_UNDERSTAND_THAT_TELNET_IS_INSECURE))
      $(shell $(COLOR_ECHO) "$(COLOR_RED)Telnet will be started automatically, "\
                            "make sure you understand why this almost certainly "\
                            "is a REALLY BAD idea before proceeding!$(COLOR_RESET)" 1>&2)
      $(error I_UNDERSTAND_THAT_TELNET_IS_INSECURE must be set to 1 to proceed)
    endif
  endif
```

---

Or maybe better just disable the `$(error)` part on `I_UNDERSTAND_THAT_TELNET_IS_INSECURE == 1` and keep the warning.

---

won't this make the example fail in murdock?

---

The example could set `I_UNDERSTAND_THAT_TELNET_IS_INSECURE = 1` in the `Makefile`